File: Difficulty: Easy/Square root of a number/square-root-of-a-number.py
#User function Template for python3


#Complete this function
class Solution:
    def floorSqrt(self, x): 
        lo=1
        hi=x
        ans=-1
        while(lo<=hi):
            mid=(lo+hi)//2
            if mid*mid==x:
                return mid
            elif (mid*mid>x):
                hi=mid-1
            else:
                ans=mid
                lo=mid+1
        return ans
                
                
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math



def main():
        T=int(input())
        while(T>0):
            
            x=int(input())
            
            print(Solution().floorSqrt(x))
            
            T-=1


if __name__ == "__main__":
    main()
# } Driver Code Ends

File: Difficulty: Easy/Square root of a number/square-root-of-a-number.java
//{ Driver Code Starts
import java.util.Scanner;

class SquartRoot {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            long a = sc.nextInt();
            Solution obj = new Solution();
            System.out.println(obj.floorSqrt(a));
            t--;
        }
    }
}
// } Driver Code Ends


class Solution {
    long floorSqrt(long n) {
        long ans=-1;
        long low=1;
        long high=n;
        while(low<=high){
            long mid=low+(high-low)/2;
            if(mid*mid<=n){
                ans=mid;
                low=mid+1;
            }else{
                high=mid-1;
            }
        }
        return ans;
    }
}
